### PR TITLE
feat: Add default namespace in the sync store layer for FS implementations

### PR DESCRIPTION
**Pull request by yquansah**
Add the ability for the `syncedStore` to default to the "default" namespace if the namespace key was not passed in by the client. 

This is done at the `syncStore` layer to cover all FS implementations. We originally thought to do this at the server layer but it seems that a "default" namespace is a storage level concern.

We could very well swap in a storage backend that we do not want to be "default" namespace aware.

Completes FLI-629

### CODE REVIEW

File: internal/storage/fs/sync.go [hunk 1]
```diff
@@ -22,69 +22,109 @@ func (s *syncedStore) GetFlag(ctx context.Context, namespaceKey string, key stri
 	s.mu.RLock()
 	defer s.mu.RUnlock()
 
+	if namespaceKey == "" {
+		namespaceKey = flipt.DefaultNamespace
+	}
```
Comment (markphelps) [inline internal/storage/fs/sync.go:27]: would you mind adding tests for `default` namespace for fs stores?

Comment (yquansah) [reply internal/storage/fs/sync.go:27]: @markphelps Yeah let me do that. Apologies!

**Review by GeorgeMac — APPROVED** [review]
The change looks good.

One thing I would be curious about is what happens if you drop these lines from the integration test:
https://github.com/flipt-io/flipt/blob/06f18d2d04cfa918004768f81ab296f7b343ac65/build/testing/integration/integration.go#L51-L53
If your change would fail before and pass after.

I was curious why we dont pick this kind of thing up in ITs and it seems we always use `default` instead of `""` because of those lines.

**Review by markphelps — APPROVED** [review]
Lgtm!

**Pull request merged**